commit 1787041a591aa84fa5e4001200b843fef4cf36cc (refs/changes/55/2055/5)
Author: Cody Peter Mello <cody.mello@joyent.com>
Date:   2017-06-08T19:14:54+00:00 (2 years, 4 months ago)
    
    NAPI-387 Integration tests should not fail tests when fabrics aren't setup
    Reviewed by: Julien Gilli <julien.gilli@joyent.com>
    Approved by: Trent Mick <trent.mick@joyent.com>

diff --git a/test/lib/fabrics.js b/test/lib/fabrics.js
index e785109..d462eed 100644
--- a/test/lib/fabrics.js
+++ b/test/lib/fabrics.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2015, Joyent, Inc.
+ * Copyright (c) 2017, Joyent, Inc.
  */
 
 /*
@@ -14,6 +14,7 @@
 
 'use strict';
 
+var assert = require('assert-plus');
 var h = require('../integration/helpers');
 var test = require('tape');
 
@@ -25,6 +26,7 @@ var test = require('tape');
 
 var ENABLED = false;
 var CHECKED = false;
+var SKIP_TESTS = (process.env.NAPITEST_SKIP_FABRICS === 'true');
 
 
 
@@ -33,7 +35,8 @@ var CHECKED = false;
  */
 function checkIfEnabled(callback) {
     if (CHECKED) {
-        return setImmediate(callback, null, ENABLED);
+        setImmediate(callback, null, ENABLED);
+        return;
     }
 
     var client = h.createNAPIclient();
@@ -41,26 +44,17 @@ function checkIfEnabled(callback) {
         CHECKED = true;
 
         if (err) {
-            return callback(err);
+            callback(err);
+            return;
         }
 
-        if (res.config && res.config.fabrics_enabled) {
-            ENABLED = true;
-        }
-
-        return callback(null, ENABLED);
-    });
-}
+        assert.object(res.config, 'res.config');
+        assert.bool(res.config.fabrics_enabled, 'res.config.fabrics_enabled');
 
+        ENABLED = res.config.fabrics_enabled;
 
-/**
- * A failing test that is run instead of the regular one if fabrics aren't
- * enabled.  This is because if you call `test.skip()`, no output at all is
- * generated, so the fabrics tests would have no output and exit 0.
- */
-function failFabricsTest(t) {
-    t.fail('fabrics not enabled - skipping test');
-    return t.end();
+        callback(null, ENABLED);
+    });
 }
 
 
@@ -70,21 +64,25 @@ function failFabricsTest(t) {
  * checkIfEnabled() runs first), these tests will show up *after* the other
  * high-level tests run.
  */
-function testIfFabricsEnabled(/* desc, [opts], next */) {
-    var testArgs = arguments;
-    var testName = Array.prototype.slice.call(arguments, 0, 1)[0];
-
-    checkIfEnabled(function _afterCheck(err, enabled) {
-        if (err) {
-            test.comment('ping error: ' + JSON.stringify(err.body));
-            return test(testName, failFabricsTest);
-        }
-
-        if (enabled) {
-            return test.apply(null, testArgs);
-        } else {
-            return test(testName, failFabricsTest);
-        }
+function testIfFabricsEnabled(testName, testFunc) {
+    assert.string(testName, 'testName');
+    assert.func(testFunc, 'testFunc');
+
+    test(testName, function (t) {
+        checkIfEnabled(function _afterCheck(err, enabled) {
+            if (err) {
+                t.ifErr(err, 'ping error - skipping test');
+                t.end();
+                return;
+            }
+
+            if (enabled) {
+                testFunc(t);
+            } else {
+                t.ok(SKIP_TESTS, 'fabrics not enabled - skipping test');
+                t.end();
+            }
+        });
     });
 }
 
